src/custom-elements/src/components/date-picker/date-picker.tsx

```diff
@@ -462,7 +516,7 @@ export class DatePicker {
             type="text"
             placeholder={this.mode === 'datetime' ? `${this.format}, --:-- --` : this.format}
             ref={(el) => (this._input = el as HTMLInputElement)}
-            value={this.value && this.multiple ? String(this.value).replace(/,/g, ', ') : this.value}
+            value={pending24hrConvertion ? splitDate : this.value && this.multiple ? String(this.value).replace(/,/g, ', ') : this.value}
```

Comment:
Way to avoid uncomfortable visual jumps in input value e.g. time is 18:15, then quickly converts to 06:15 pm. Room for improvement.